administrator/components/com_jedchecker/libraries/rules/jamss.php

```diff
@@ -400,11 +400,7 @@ private function scan_file($path)
 	 */
 	private function calculate_line_number($offset, $file_content)
 	{
-		// Fetches all the text before the match
-		list($first_part) = str_split($file_content, $offset);
-		$line_nr = strlen($first_part) - strlen(str_replace("\n", "", $first_part)) + 1;
-
-		return $line_nr;
+		return substr_count($file_content, "\n", 0, $offset) + 1;
```

Comment:
This is a simpler way, yes... but it is harder to understand for you average developer since your using a function that is 
partially being miss matched with in the actual variable naming.
Your Code
```php
substr_count($file_content, "\n", 0, $offset) + 1;
```
Documentation:
```php
substr_count ($haystack, $needle, $offset, $length);
```
Now I can understand what you are doing, but in fact `$offset` is no longer true to its intent. So we need to rename and better document what is being done here.

Then for the sake of PHP 8 we should start renaming all function variable to camelCase. This is what I think your code should look like:
```php
/**
 * Calculates the line number where pattern match was found
 *
 * @param   int  $length        The maximum length after the specified offset to search for the line breaks
 * @param   string  $fileContent  The file content in string format
 * @param   int  $offset  The offset where to start counting (default 0)
 *
 * @return  int  Returns        line number where the subject code was found
 */
private function calculate_line_number($length, $fileContent, $offset = 0)
{
	// we are counting the number of line breaks between 
	// the offset, and length in the file content.
	// Then we add one since we get a zero
	// based return value from substr_count function
	return substr_count($fileContent, "\n", $offset, $length) + 1;
}
```
This way any who look at this method knows exactly what it does. Please make the changes and this can merge.

Reply:
Well maybe you don't need to add the offset, we can hard code it to 0. But with explanation as much as possible.

Reply:
In favor of the changes proposed by @Llewellynvdm please update your PR, so it can be merged before the new release.